From 0fa9c1c2fe923ca4f36573ca6e6b97e555d0802d Mon Sep 17 00:00:00 2001
From: mokopatches <mokopatches@openmoko.org>
Date: Wed, 16 Jul 2008 14:44:10 +0100
Subject: [PATCH] explicitly-link-notes-section.patch

Since 2.6.23 kbuild produces a 3GB arch/arm/boot/Image because it includes a
.note.gnu.build-id section at address 0 which is followed by 3GB of 0x00.
The --build-id option is set in the toplevel Makefile.
This patch explicitly puts the notes section after the TEXT section.
---
 arch/arm/kernel/vmlinux.lds.S |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/arch/arm/kernel/vmlinux.lds.S b/arch/arm/kernel/vmlinux.lds.S
index 4898bdc..b835564 100644
--- a/arch/arm/kernel/vmlinux.lds.S
+++ b/arch/arm/kernel/vmlinux.lds.S
@@ -106,6 +106,8 @@ SECTIONS
 		*(.got)			/* Global offset table		*/
 	}
 
+	NOTES
+
 	RODATA
 
 	_etext = .;			/* End of text and rodata section */
-- 
1.5.6.3

